backend/controllers/iepHistory.js

```diff
@@ -0,0 +1,50 @@
+const moment = require('moment')
+
+const getIepHistoryFactory = elite2Api => {
+  const getIepHistory = async (context, offenderNo) => {
+    const bookingDetails = await elite2Api.getDetails(context, offenderNo)
+    const iepSummary = await elite2Api.getIepSummaryWithDetails(context, bookingDetails.bookingId)
+    const iepHistoryDetails = []
+
+    /* eslint-disable no-restricted-syntax */
```

Comment:
The current coding style is more functional. See example https://github.com/ministryofjustice/prisonstaffhub/commit/3475ab3590df98e78b5be6197a6d5ff2180d26b8

Reply:
That's brilliant, thank you so much! I made a some small changes to make sure it only called the apis once per agency/user, but other than that it worked pretty much out of the box. Thank you so much for taking the time!